keras/layers/pooling/generalized_mean_pooling1d.py

```diff
@@ -0,0 +1,136 @@
+    array([[[1.6509637],
+            [2.596247 ],
+            [3.5700185],
+            [4.5548835]]], dtype=float32)>
+
+    2. When pool_size=2, strides=1, padding='valid'
+
+    >>> input_shape = (2, 3, 4)
+    >>> x = tf.random.normal(input_shape)
+    >>> gem_pool_1d = tf.keras.layers.GeneralizedMeanPooling1D(power=3,
+    ...    pool_size=2, strides=2, padding='valid',
+    ...    data_format='channels_last')
+    >>> gem_pool_1d(x)
+    <tf.Tensor: shape=(1, 2, 1), dtype=float32, numpy=
+    array([[[1.6509637],
+            [3.5700185]]], dtype=float32)>
+
+    3. When pool_size=2, strides=1, padding='same'
+
+    >>> input_shape = (2, 3, 4)
+    >>> x = tf.random.normal(input_shape)
+    >>> gem_pool_1d = tf.keras.layers.GeneralizedMeanPooling1D(power=3,
+    ...    pool_size=2, strides=1, padding='same',
+    ...    data_format='channels_last')
+    >>> gem_pool_1d(x)
+    <tf.Tensor: shape=(1, 5, 1), dtype=float32, numpy=
+    array([[[1.6509637],
+            [2.596247 ],
+            [3.5700185],
+            [4.5548835],
+            [5.0000005]]], dtype=float32)>
+
+    Args:
+      power: Float power > 0 is an inverse exponent parameter, used during
+        the generalized mean pooling computation. Setting this exponent as
+        power > 1 increases the contrast of the pooled feature map and focuses
+        on the salient features of the image. GeM is a generalization of the
+        average pooling when `power` = 1 and of spatial max-pooling layer when
+        `power` = inf or a large number.
+      pool_size: Integer, size of the average pooling windows.
+      strides: Integer, or None. Factor by which to downscale.
+        E.g. 2 will halve the input.
+        If None, it will default to `pool_size`.
+      padding: A string. The padding method, either 'valid' or 'same'.
+        `'valid'` means no padding. `'same'` results in padding evenly to
+        the left/right or up/down of the input such that output has the same
+        height/width dimension as the input.
+      data_format: A string, one of `channels_last` (default) or
+        `channels_first`. The ordering of the dimensions in the inputs.
+        `channels_last` corresponds to inputs with shape
+        `(batch, steps, features)` while `channels_first` corresponds
+        to inputs with shape `(batch, features, steps)`.
+      name: A string, the name of the layer.
+
+    Input shape:
+      - If `data_format='channels_last'`:
+        3D tensor with shape `(batch_size, steps, features)`.
+      - If `data_format='channels_first'`:
+        3D tensor with shape `(batch_size, features, steps)`.
+
+    Output shape:
+      - If `data_format='channels_last'`:
+        3D tensor with shape `(batch_size, downsampled_steps, features)`.
+      - If `data_format='channels_first'`:
+        3D tensor with shape `(batch_size, features, downsampled_steps)`.
+
+     References:
+        - https://arxiv.org/pdf/1711.02512.pdf
```

Comment:
Format this as a markdown link with paper title (or author names), linking to the abstract not the PDF.